master/internal/trial.go

```diff
@@ -381,6 +381,33 @@ func (t *trial) maybeAllocateTask() error {
 	}
 
 	restoredAllocation, err := t.maybeRestoreAllocation()
+
+	if restoredAllocation == nil {
+		launchWarnings, err := t.rm.ValidateResourcePoolAvailability(
```

Comment:
can we put this inside another function?

we should also call it after we check the err from ```t.maybeRestoreAllocation```